File: app/src/androidTest/java/net/tlalka/android/fiszki/test/components/TestApplicationComponent.java
package net.tlalka.android.fiszki.test.components;

import javax.inject.Singleton;

import dagger.Component;
import net.tlalka.android.fiszki.core.modules.ApplicationModule;
import net.tlalka.android.fiszki.core.modules.SessionModule;

@Singleton
@Component(modules = ApplicationModule.class)
public interface TestApplicationComponent {

    TestSessionComponent add(SessionModule sessionModule);
}

File: app/src/androidTest/java/net/tlalka/android/fiszki/test/components/TestApplicationComponent.kt
package net.tlalka.android.fiszki.test.components

import dagger.Component
import net.tlalka.android.fiszki.core.modules.ApplicationModule
import net.tlalka.android.fiszki.core.modules.SessionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [ApplicationModule::class])
interface TestApplicationComponent {

    fun add(sessionModule: SessionModule): TestSessionComponent
}